File: LifeTrackerFullStack/LifeTrackerBackend/Models/GameTask.cs
namespace LifeTracker.Models;

public enum Difficulty
{
    Easy,
    Medium,
    Hard
}

public class GameTask
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Difficulty Difficulty { get; set; }
    public bool IsCompleted { get; set; } = false;
    public int RewardXP { get; set; }
}

File: LifeTrackerMobile/app/src/main/java/com/lifetracker/mobile/data/GameTask.kt
package com.lifetracker.mobile.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "tasks")
data class GameTask(
    @PrimaryKey(autoGenerate = true)
    val localId: Long = 0,

    @SerializedName("id")
    val serverId: Long? = null,

    val title: String,
    val description: String? = null,
    @SerializedName("rewardXp")
    val rewardXp: Int,
    val isCompleted: Boolean = false,
    val isSynced: Boolean = false
)

